control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -753,6 +772,14 @@ impl SpecOperationsHelper for VolumeSpec {
             }
 
             VolumeOperation::SetReplica(replica_count) => {
+                // Validation for volume group volume's replica count cannot go below 2.
+                if self.volume_group.is_some() && *replica_count < 2 {
```

Comment:
This should go below, otherwise if replica count is already 1, it doesn't fail with the ReplicaCountAchieved error.